WeakAuras/WeakAuras.lua

```diff
@@ -3322,6 +3365,17 @@ local function pAdd(data)
   end
 
   db.displays[id] = data;
+
+  data.uid = data.uid or WeakAuras.GenerateUniqueID()
+  local otherID = UIDtoID[data.uid]
+  if not otherID then
+    UIDtoID[data.uid] = id
+  elseif otherID ~= id then
+    -- duplicate uid
+    data.uid = WeakAuras.GenerateUniqueID()
```

Comment:
This will cause a collision with approximately a 1 in 73 quintillion chance. If someone legitimately does get a collision like this, then we should award the person who makes the bug report a prize ;)